lib/std/private/decode_helpers.nim

```diff
@@ -1,9 +1,25 @@
-proc handleHexChar*(c: char, x: var int, f: var bool) {.inline.} =
+proc handleHexChar*(c: char, x: var int): bool {.inline.} =
+  ## Converts `%xx` hexadecimal to the ordinal number and adds the result to `x`.
+  ## Returns `true` if `c` is hexadecimal.
+  ##
+  ## When `c` is hexadecimal, the proc is equal to `x = x shl 4 + hex2Int(c)`.
+  runnableExamples:
+    var x: int = 0
+    assert handleHexChar('a', x)
+    assert x == 10
+
+    assert handleHexChar('B', x)
+    assert x == 171 # 10 << 4 + 11
+
+    # failed
+    assert not handleHexChar('?', x)
```

Comment:
add:
```nim
assert x == 171 # unchanged
```
(examples are part of documentation, no need to clarify that in doc comment, it'd be redundant with this)